### PR TITLE
New Detector: boolean variable `isNotXXX` instead of `isXXX` 

### PR DESCRIPTION
**Pull request by TilakMaddy**
This detector discourages variable names like `isNotPaused` , `isNotRewarded` , `doesNotMint` etc  and encourages to change them to `isPaused`, `isRewarded` , etc

This practice will cause lesser confusion in the codebase and possibly avoid future mistakes

### CODE REVIEW

File: aderyn_core/src/detect/nc/bool_is_not.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+            if let Some(TypeName::ElementaryTypeName(e)) = &variable_declaration.type_name {
+                if e.name == "bool"
+                    && (var_name_lowercase.starts_with("isnot")
+                        || var_name_lowercase.starts_with("doesnot"))
+                {
+                    capture!(self, context, variable_declaration);
+                }
+            }
+        }
```
Comment (TilakMaddy) [inline aderyn_core/src/detect/nc/bool_is_not.rs:34]: Logic here

File: tests/contract-playground/src/BooleanIsNot.sol [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+contract BooleansIsNot {
+
+    bool public isHappy = true;
+    bool private isNotEnjoying;
+    bool public doesWork = true;
+    bool public doesNotDoHomework;
```
Comment (TilakMaddy) [inline tests/contract-playground/src/BooleanIsNot.sol:9]: example here

**Comment by alexroan** [comment]
I'm not convinced this is worthy of being something that Aderyn recommends as a security concern right now. Let's shelf this.

**Comment by TilakMaddy** [comment]
@alexroan  This one was also found in Cantina's bot report for GOAT.TECH under the heading incorrect bool naming convention :P 

I agree though, this is not a high quality one ☝️

**Comment by TilakMaddy** [comment]
Link to report - https://gist.github.com/ChaseTheLight01/d61a9fd3791763e8b487961bf75887de
Link to competition - https://cantina.xyz/competitions/f214cf86-cc80-40c0-a70b-e9bb25d7ac80

**Pull request closed without merging**